bindings/pyroot_experimental/pythonizations/python/ROOT/_application.py

```diff
@@ -73,9 +73,30 @@ def displayhook(v):
     def init_graphics(self):
         """Configure ROOT graphics to be used interactively"""
 
+        # Note that we only end up in this function if gROOT.IsBatch() is false
+        import __main__
         if self._is_ipython and 'IPython' in sys.modules and sys.modules['IPython'].version_info[0] >= 5:
+            # ipython and notebooks, register our event processing with their hooks
             self._ipython_config()
-        else:
+        elif sys.flags.interactive == 1 or not hasattr(__main__, '__file__'):
+            # Python in interactive mode, use the PyOS_InputHook to call our event processing
+            # - sys.flags.interactive checks for the -i flags passed to python
+            # - __main__ does not have the attribute __file__ if the Python prompt is started directly
             self._inputhook_config()
+        else:
+            # Python in script mode, start a separate thread for the event processing
+            def process_root_events():
+                while True:
```

Comment:
Perhaps we could still use a condition here to keep looping, and that condition is set to False in the atexit handler, as it is done for old PyROOT:
https://github.com/root-project/root/blob/master/bindings/pyroot/ROOT.py#L853

Reply:
Plus joining the thread afterwards.

Reply:
Added!